pkg/relay/ethereum/tx.go

```diff
@@ -70,6 +70,7 @@ func (c *Chain) SendMsgs(msgs []sdk.Msg) ([]core.MsgID, error) {
 			logger.Error("failed to get receipt", err, "msg", msg)
 			return nil, err
 		} else if receipt.Status == gethtypes.ReceiptStatusFailed {
+			logger.Error("tx execution failed", fmt.Errorf("revertReason=%s, msgIndex=%d", revertReason, i), "msg", msg)
```

Comment:
I have a simple question. Why did you move "msg" from the error to the log attributes and leave "revertReason" and "msgIndex" in it?

Reply:
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/13/files/4611488cb01d764157417b2ccbf882ec9e47d78f..2f9a74a137bf324cfb11b70b760d9b0313b96a4e#diff-26026a3e1f9975f0f18cd6cec78a84c6a01ef33c3ee7d38af0bca97a08adb9ceL62

https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/13/files/4611488cb01d764157417b2ccbf882ec9e47d78f..2f9a74a137bf324cfb11b70b760d9b0313b96a4e#diff-26026a3e1f9975f0f18cd6cec78a84c6a01ef33c3ee7d38af0bca97a08adb9ceL66

This same pattern was used to output msg, revertReason, msgIndex to err.

Reply:
@dongrie In terms of structured logging, any useful values should be embedded as attributes.
On the other hand, it is also good to embed much information in errors.
Therefore, I think all attributes (revertReason, msg, msgIndex) should be embedded in both.
It would be a bit redundant, but seems most reasonable to me.
What do you think about it?

Reply:
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/13/commits/2d359b6da1fd988b1ad4a65bd8e5e96009fe616d
msg is now included in err